src/Microsoft.AspNet.Server.Kestrel/Infrastructure/Constants.cs

```diff
@@ -20,5 +24,164 @@ internal class Constants
+        internal static readonly byte[] HeaderBytesStatus426 = Encoding.ASCII.GetBytes("426 Upgrade Required");
+        internal static readonly byte[] HeaderBytesStatus500 = Encoding.ASCII.GetBytes("500 Internal Server Error");
+        internal static readonly byte[] HeaderBytesStatus501 = Encoding.ASCII.GetBytes("501 Not Implemented");
+        internal static readonly byte[] HeaderBytesStatus502 = Encoding.ASCII.GetBytes("502 Bad Gateway");
+        internal static readonly byte[] HeaderBytesStatus503 = Encoding.ASCII.GetBytes("503 Service Unavailable");
+        internal static readonly byte[] HeaderBytesStatus504 = Encoding.ASCII.GetBytes("504 Gateway Timeout");
+        internal static readonly byte[] HeaderBytesStatus505 = Encoding.ASCII.GetBytes("505 HTTP Version Not Supported");
+        internal static readonly byte[] HeaderBytesStatus506 = Encoding.ASCII.GetBytes("506 Variant Also Negotiates");
+        internal static readonly byte[] HeaderBytesStatus507 = Encoding.ASCII.GetBytes("507 Insufficient Storage");
+        internal static readonly byte[] HeaderBytesStatus510 = Encoding.ASCII.GetBytes("510 Not Extended");
+
+        internal static readonly byte[] HeaderBytesConnectionClose = Encoding.ASCII.GetBytes("\r\nConnection: close");
+        internal static readonly byte[] HeaderBytesConnectionKeepAlive = Encoding.ASCII.GetBytes("\r\nConnection: keep-alive");
+        internal static readonly byte[] HeaderBytesTransferEncodingChunked = Encoding.ASCII.GetBytes("\r\nTransfer-Encoding: chunked");
+        internal static readonly byte[] HeaderBytesHttpVersion1_0 = Encoding.ASCII.GetBytes("HTTP/1.0 ");
+        internal static readonly byte[] HeaderBytesHttpVersion1_1 = Encoding.ASCII.GetBytes("HTTP/1.1 ");
+        internal static readonly byte[] HeaderBytesContentLengthZero = Encoding.ASCII.GetBytes("\r\nContent-Length: 0");
+        internal static readonly byte[] HeaderBytesSpace = Encoding.ASCII.GetBytes(" ");
+        internal static readonly byte[] HeaderBytesServer = Encoding.ASCII.GetBytes("\r\nServer: Kestrel");
+        internal static readonly byte[] HeaderBytesDate = Encoding.ASCII.GetBytes("Date: ");
+        internal static readonly byte[] HeaderBytesEndHeaders = Encoding.ASCII.GetBytes("\r\n\r\n");
+
+        internal static readonly byte[] HexBytes = Encoding.ASCII.GetBytes("0123456789abcdef");
+
+        internal static readonly byte[] HeaderEndChunkBytes = Encoding.ASCII.GetBytes("\r\n");
+        internal static readonly byte[] HeaderEndChunkedResponseBytes = Encoding.ASCII.GetBytes("0\r\n\r\n");
+        internal static readonly byte[] HeaderContinueBytes = Encoding.ASCII.GetBytes("HTTP/1.1 100 Continue\r\n\r\n");
+        internal static readonly byte[] HeaderEmptyDataBytes = new byte[0];
+
+        // method needed while this bug is active https://github.com/dotnet/coreclr/issues/2526
+        public static void InitializeJitConstants()
+        {
```

Comment:
I chatted w/ @jkotas, and he suggested this could be simplified to:

    private static bool initializer;

    public static void InitializeJitConstants()
    {
         initializer = BitConverter.IsLittleEndian;
    }

All static fields of a class get initialized at once.

He also suggested it's better to avoid static variables altogether. This could probably be achieved by making `Constants` non-static and referencing the singleton instance using a property in `ServiceContext`. I think we should probably do this instead, so we don't need to worry about static initialization order.